src/components/VLink.vue

```diff
@@ -0,0 +1,58 @@
+<template>
+  <Component :is="linkComponent" v-bind="linkProperties" v-on="$listeners"
+    ><slot
+  /></Component>
+</template>
+
+<script>
+/**
+ * This is a wrapper component for all links. All VLinks have to have a valid `href` attr.
+ * Links with `href` starting with `/` are treated as internal links.
+ *
+ * Internal links use `NuxtLink` component with `to` attribute set to `localePath(href)`
+ * External links use `a` element set to open in a new tab and not raise an error with the current iframe setup.
+ */
+import {
+  computed,
+  defineComponent,
+  useContext,
+  watch,
+} from '@nuxtjs/composition-api'
+import { warn } from '@/utils/warn'
+
+const defaultProps = { target: '_blank', rel: 'noopener noreferrer' }
+
+export default defineComponent({
+  name: 'VLink',
+  setup(_, { attrs }) {
+    const { app } = useContext()
+    const isInternal = computed(() => attrs.href?.startsWith('/'))
+    const linkComponent = computed(() => (isInternal.value ? 'NuxtLink' : 'a'))
+
+    let linkProperties = {}
+    watch(
+      [attrs],
```

Comment:
Does watching `attrs` like this actually work? :thinking: When I test this locally it doesn't seem to fire the watcher when the attrs change (even though it does re-render). You can see in this example where clicking the button causes the re-render but doesn't fire the watcher to update the value: https://codesandbox.io/s/crazy-rosalind-y12vp4

I think attrs is probably the wrong tool for this anyway... `href` should just be a required string prop, and then we can add the `warn` in the validator function for it.

`watch` _will_ work on props, so we can successfully keep linkProperties updated (in fact, it can just be a `computed` at that point.

Reply:
Right! Having href as a prop will ensure that it's not also passed as an attr to NuxtLink! So, there will be no need to destructure the attrs.